src/main/steps/index.ts

```diff
@@ -41,13 +48,19 @@ export const getNextStepUrl = (req: AppRequest, data: Partial<Case>): string =>
     return CITIZEN_HOME_URL;
   }
   const { path, queryString: queryStr } = getPathAndQueryString(req);
-  const nextStep = [
+  let nextStep = [
     ...citizenSequence,
     ...respondentCaseSequence,
     ...applicantCaseSequence,
     ...responseCaseSequence,
     ...C100Sequence,
   ].find(s => s.url === path);
+  if (
```

Comment:
May I know why this condition is required?

Reply:
Find is always getting the steps for respondent/keep-details-private from respondentCaseSequence because it is checked before responseCaseSequence. The condition is used to get the step from responseCaseSequence instead, if the user has navigated from tasklistresponse.

Reply:
Please move this condition inside respondentCaseSequence and not in generic file.

Reply:
Condition moved to respondentCaseSequence